src/hal/src/device.rs

```diff
@@ -278,12 +291,17 @@ pub trait Device<B: Backend>: Any + Send + Sync {
     /// which references the framebuffer, has finished execution.
     fn destroy_framebuffer(&self, buf: B::Framebuffer);
 
+    /// Creates a new shader module object through the SPIR-V binary data
```

Comment:
let's be a little more consistent: `Create` instead of `Creates`. Also - dots at the end of sentences.